Makefile

```diff
@@ -301,6 +301,14 @@ test:
 	cd awxkit && $(VENV_BASE)/awx/bin/tox -re py3
 	awx-manage check_migrations --dry-run --check  -n 'missing_migration_file'
 
+github_ci_setup:
+	echo $(CI_GITHUB_TOKEN) | docker login ghcr.io -u $(GITHUB_ACTOR) --password-stdin  # Log in to registry
+	docker pull $(DEVEL_IMAGE_NAME) || :  # Pre-pull image to warm build cache
+	make docker-compose-build	 # Build image
+
+github_ci_runner: github_ci_setup
```

Comment:
I'm wondering if this one might be a good "Generic" command to have for reuse? i.e. just call it container_runner or something, change `GI_GITHUB_COMMAND` to just `COMMAND_TO_RUN ?= awx-manage shell_plus` and also swap out `GITHUB_WORKSPACE` to something like `MOUNT_DIR ?= .`

Reply:
There's a messy mess of environment variables here, let me categorize:
 - Things that start with `CI_`  had to be added specifically for this makefile target. This is because they were in the github workflow varspace, but not as env vars, like `${{ secrets.GITHUB_TOKEN }}`, so the github actions has to add it as a (novel) env var to pass the data
 - A couple of things like `GITHUB_WORKSPACE` are documented to be provided automatically by github actions, so there's no need to pass anything, but we also don't have freedom to rename it, as long as we want to take advantage of it. That's why I wouldn't want to call it `MOUNT_DIR`
 - Many, many, things are already declared Makefile variables like `COMPOSE_TAG`, so I used those existing names.

Sure, it could be used for other uses, but I'd like to be very explicit that the first and dominating purpose is to use it for Github actions. You have a really helpful suggestion about the help text and I'll look into that, because otherwise what I wrote here about the env vars will be promptly forgotten.

Reply:
I tried to give this some genuine thought, so here's my answer -

I still want dedicated `github_` targets, but this is an oddly general target. Thus, I split out a new command `make docker-runner` which should be compared to `make docker-compose` and others. Then, `github_ci_runner` is just a combination of the github setup command & ansible-runner.